### ISSUE TITLE
aws-amplify@^1.0.8 ERROR in ./node_modules/aws-sdk/lib/event_listeners.js Module not found: Error: Can't resolve 'util' in '\node_modules\aws-sdk\lib' 

### ISSUE THREAD

**Issue by rezaunnabi2018** [aws-amplify/amplify-js#1560]
Do you want to request a feature or report a bug?
   Report a bug 
What is the current behavior?
   At project startup it throws following error:
   ERROR in ./node_modules/aws-sdk/lib/event_listeners.js
   Module not found: Error: Can't resolve 'util' in 'F:\WiptimeNew\node-10.3.0-wiptime\src\node-10.3.0-wiptimeclient\node_modules\aws-sdk\lib'
 @ ./node_modules/aws-sdk/lib/event_listeners.js 522:21-36
 @ ./node_modules/aws-sdk/lib/core.js
 @ ./node_modules/aws-sdk/browser.js
 @ ./node_modules/@aws-amplify/core/lib/Facet.js
 @ ./node_modules/@aws-amplify/core/lib/index.js
 @ ./node_modules/@aws-amplify/auth/lib/index.js
 @ ./src/MainApp.js
 @ ./src/index.js
 @ multi (webpack)-dev-server/client?http://0.0.0.0:3000 webpack/hot/dev-server babel-polyfill webpack/hot/only-dev-server react-hot-loader/patch ./index.js
webpack: Failed to compile.

If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem. Your bug will get fixed much faster if we can run your code and it doesn't have dependencies other than AWS Amplify.

   1. Add the module

    yarn add aws-amplify  // This line installed "^1.0.8" in my case.

   2. Import and used

    import Amplify from "aws-amplify";
    
   3. Start the project
   
    yarn start // the start command used in package.json- "start": " webpack-dev-server --hot --inline --progress --host 0.0.0.0"

What is the expected behavior?
    AWS amplify should work when imported.
Which versions of Amplify, and which browser / OS are affected by this issue? Did this work in previous versions?
    aws-amplify - 1.0.8 
    react.js version- 16.2.0
You can turn on the debug mode to provide more info for us by setting ```window.LOG_LEVEL = 'DEBUG';``` in your app.

**Comment by powerful23** [comment]
@rezaunnabi2018 can you try the latest version of aws-amplify, we just updated the aws-sdk version in our package and that may fix your problem.

**Comment by powerful23** [comment]
@markclayton can you show your `package.json` and also can you check if there are `aws-sdk` packages under `./node_modules/@aws-amplify/{category}/node_modules`

**Comment by simonemazzoni** [comment]
Same problem for me.
Any news about the issue?

@powerful23 I checked the `node_modules` folder as you pointed out but there is no such folder in `@aws-amplify/auth` folder.

**Comment by simonemazzoni** [comment]
@mlabieniec I'm not including explicitly the `aws-sdk` since I think it is already included by `@aws-amplify`. Anyway I tried also to include that package explicitly but nothing changed.

**Comment by mlabieniec** [comment]
Can you check your node_modules/aws-sdk and see what version it is. Or I think npm info command might tell you also. How did you install amplify? Via modules like npm install @aws-amplify/auth or global npm install aws-amplify?

**Comment by simonemazzoni** [comment]
@mlabieniec I actually use yarn, which shouldn't change anything.
This is the installed `aws-sdk` version (which looks like it is the latest)
`yarn list v1.10.1
└─ aws-sdk@2.333.0
`

Notice that I'll install `@aws-aplify/auth` using `yarn add`, but installing the complete `@aws-amplify` package doesn't change the result.

**Comment by simonemazzoni** [comment]
No I didn't use create-react-app. I could try to setup a test app using create react, but I don't think it will resolve the issue.

My setup uses babel7 also. I don't know if it can be part of the reason of the error.

**Comment by mlabieniec** [comment]
Hm yes this is likely the issue since it’s failing at the compile stage. I don’t think the aws-sdk compiles to this target right now (assuming that’s es next or 7)

**Comment by simonemazzoni** [comment]
@mlabieniec I see that there were 3 updates to the `aws-sdk` package since 5 days ago. Assuming that, as you said the problem is in `aws-sdk`, maybe could be fixed?

**Comment by d-olivas** [comment]
I was having the same issue in my react project. What fixed it for me was to update the node version I am using, I was using node 8.11.4 and updated to 11.3.0. What was causing the problem was this line `var params = require('util').inspect(censoredParams, true, null);` in `./node_modules/aws-sdk/lib/event_listeners.js` the problem was that the require was missing './' and that did't allow the app to compile properly. I'm using aws-amplify v1.1.19 and it has as a dependency aws-sdk v2.329.0 and currently aws-sdk is in version 2.411.0... Anyways, I don't know why that require had missing the './' and I just guessed that maybe a newer version of node would interpret that, and it worked. I looked again in `./node_modules/aws-sdk/lib/event_listeners.js` and now the require has the './', so honestly I don't know exactly what happened there, but now it works. Hope someone can find this helpful some day.
